### PR TITLE
Run integration tests for additional DNS providers

### PR DESCRIPTION
**Pull request by MaxHorstmann**
Follow-up to https://github.com/StackExchange/dnscontrol/pull/928: add

* Azure DNS
* NAMEDOTCOM
* GCloud
* Cloudflare

([secrets](https://github.com/StackExchange/dnscontrol/settings/secrets))

### CODE REVIEW

**Comment by tlimoncelli** [comment]
The integration tests can run in parallel (they all use different scratch DNS domains).  Can github actions do that?

**Comment by MaxHorstmann** [comment]
GH can't run steps in parallel *yet* ([common request](https://github.community/t/steps-in-parallel/16343)). It can run jobs in paralllel. I'll check how hard it would be to refactor this into separate jobs.

**Comment by MaxHorstmann** [comment]
@tlimoncelli @atma-stackoverflow this works great (faster than TC due to parallel execution!) and is another nice shippable piece. Mind taking a quick look?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -5,34 +5,59 @@ on:
...
-      working-directory: integrationTest
-      run: go test -v -verbose -provider BIND
-
-    - name: Run integration test for ROUTE53 provider
+  integration-tests:
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      matrix:
+        provider: [BIND, AZURE_DNS, NAMEDOTCOM, ROUTE53, GCLOUD, CLOUDFLAREAPI]
```
Comment (tlimoncelli) [inline .github/workflows/build.yml:30]: Minor nit.  Sort the names of the providers and the list of env variables.

**Review by tlimoncelli — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (MaxHorstmann) [reply .github/workflows/build.yml]: ok! https://github.com/StackExchange/dnscontrol/commit/f145ae14386f92a04408851e5f5efaa725e898fe

I originally kept them in the same order as in TC so they would run in the same sequence. But they're running in parallel now, so there's no sequence and this is a tad more readable.

**Pull request merged**